src/python/pants/backend/python/subsystems/lambdex.py

```diff
@@ -20,6 +30,44 @@ class Lambdex(PythonToolBase):
     default_lockfile_resource = ("pants.backend.python.subsystems", "lambdex.lock")
     lockfile_rules_type = LockfileRules.SIMPLE
 
+    layout = EnumOption(
+        default=LambdexLayout.LAMBDEX,
+        help=softwrap(
+            """
+            Explicitly control the layout used for `python_awslambda` and
+            `python_google_cloud_function` targets. This option exists for the transition from
+            Lambdex-based layout to the plain zip layout, as recommended by cloud vendors.
+            """
+        ),
+    )
+
+    def warn_for_layout(self, target_alias: str) -> None:
+        if self.options.is_default("layout"):
+            lambda_message = (
+                " (you will need to also update the handlers configured in the cloud from `lambdex_handler.handler` to `lambda_function.handler`)"
+                if target_alias == "python_awslambda"
+                else ""
+            )
+
+            warn_or_error(
+                "2.19.0.dev0",
+                f"using the Lambdex layout for `{target_alias}` targets",
+                softwrap(
+                    f"""
+                    Set the `[lambdex].layout` option explicitly to `zip` (recommended) or `lambdex`
+                    (compatibility), in `pants.toml`. Recommended: set to `zip` to opt-in to the new
```

Comment:
Should mention here that the exported handler function name changes when you set to `zip`?

Reply:
Yeah, for sure. I think that's covered by `lambda_message` since it only applies to AWS, not GCF.

Reply:
When I run this in my work repo, I get:

> 12:02:26.94 [WARN] DEPRECATED: using the Lambdex layout for `python_awslambda` targets is scheduled to be removed in version 2.19.0.dev0.
>
> Set the `[lambdex].layout` option explicitly to `zip` (recommended) or `lambdex` (compatibility), in `pants.toml`. Recommended: set to `zip` to opt-in to the new layout recommended by cloud vendors (you will need to also update the handlers configured in the cloud from `lambdex_handler.handler` to `lambda_function.handler`):
>
>     [lambdex]
>     layout = "zip"
>
> You can also explicitly set `layout = "lambdex"` to silence this warning and continue using the Lambdex-based layout in this release of Pants. This layout will disappear in future.

Let me know if that's acceptable.

Reply:
Perfect!